services/solr-names-updater/src/solr_names_updater/worker.py

```diff
@@ -0,0 +1,143 @@
+
+The entry-point is the **cb_subscription_handler**
+
+The design and flow leverage a few constraints that are placed upon it
+by NATS Streaming and using AWAIT on the default loop.
+- NATS streaming queues require one message to be processed at a time.
+- AWAIT on the default loop effectively runs synchronously
+
+If these constraints change, the use of Flask-SQLAlchemy would need to change.
+Flask-SQLAlchemy currently allows the base model to be changed, or reworking
+the model to a standalone SQLAlchemy usage with an async engine would need
+to be pursued.
+"""
+import json
+import os
+
+import nats
+from flask import Flask
+from namex.models import db
+from queue_common.service import QueueServiceManager
+from queue_common.service_utils import QueueException, logger
+from requests import RequestException
+from sentry_sdk import capture_message
+from sqlalchemy.exc import OperationalError
+from urllib3.exceptions import NewConnectionError
+
+from solr_names_updater import config
+from solr_names_updater.names_processors.names import (
+    process_add_to_solr as process_names_add,
+    process_delete_from_solr as process_names_delete)
+from solr_names_updater.names_processors.possible_conflicts \
+    import (process_add_to_solr as process_possible_conflicts_add,
+            process_delete_from_solr as process_possible_conflicts_delete)
+
+
+qsm = QueueServiceManager()  # pylint: disable=invalid-name
+APP_CONFIG = config.get_named_config(os.getenv('DEPLOYMENT_ENV', 'production'))
+FLASK_APP = Flask(__name__)
+FLASK_APP.config.from_object(APP_CONFIG)
+db.init_app(FLASK_APP)
+
+
+def is_nr_state_change_msg_type(msg: dict):
+    """Check message is of type nr state change."""
+    if msg and msg.get('type', '') == 'bc.registry.nr_state.changed':
+        return True
+
+    return False
+
+
+def is_nr_furnished_change_msg_type(msg: dict):
+    """Check message is of type nr furnished changed."""
+    if msg and msg.get('type', '') == 'bc.registry.nr_furnished.changed':
+        return True
+
+    return False
+
+
+def is_processable(msg: dict):
+    """Determine if message is processable using message type of msg."""
+    if msg and \
+        (is_nr_state_change_msg_type(msg)
+         or is_nr_furnished_change_msg_type(msg)):
+        return True
+
+    return False
+
+
+async def process_nr_event_message(msg: dict, flask_app: Flask):
+    """Update solr accordingly based on incoming nr state changes."""
+    if not flask_app or not msg:
+        raise QueueException('Flask App or msg not available.')
+
+    with flask_app.app_context():
+        logger.debug('entering processing of nr event msg: %s', msg)
+
+        if is_nr_state_change_msg_type(msg):
+            data = msg.get('data')
+            new_state = data.get('newState')
+            if new_state in ('APPROVED', 'CONDITIONAL'):
+                process_names_add(data)
+            elif new_state in ('CANCELLED', 'CONSUMED'):
```

Comment:
we may not want to implement `CONSUMED` logic here?  need confirmation

Reply:
I don't think consumed is a 'state' unless that is new

Reply:
no, i'm thinking not.  i just thought it was a state.

Reply:
If a name is CONSUMED it should get removed, and then added by the COLIN side of things.

Reply:
@thorwolpert Still not sure how to proceed with *CONSUME*. Where should I be adding code for publishing the “consumed” event?

Reply:
similar code as to what was added for requests, but on the name model

Reply:
Updated listener on name model to publish `CONSUMED` state change when consumption date on name is added.